enhancements/storage/csi-cinder-operator.md

```diff
@@ -0,0 +1,179 @@
+---
+title: csi-cinder-operator
+authors:
+  - "@mfedosin"
+reviewers:
+  - "@jsafrane"
+  - "@hekumar"
+  - "@chuffman"
+  - "@fbertina"
+approvers:
+  - "@..."
+creation-date: 2020-08-17
+last-updated: 2020-08-17
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# CSI Operator for Openstack Cinder
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In the past storage drivers have been provided either in-tree or via external provisioners. With the move to advocating CSI as the way to deliver storage drivers, OpenShift needs to move the in-tree drivers (and others that our customers need) to use CSI. With the focus on operators in Red Hat OCP 4.x these CSI drivers should be installed in the form of an Operator.
+
+## Motivation
+
+- In-tree drivers will be removed from Kubernetes, we need to continue to make the drivers available and CSI is the way to do this.
+
+- OpenStack is a key cloud provider for OpenShift and use of Cinder has been supported in both OpenShift 3.x and 4.x, we need to move this driver to be CSI, provided by an operator.
+
+- CSI drivers (including OpenStack Cinder) will enable new storage features, not possible with in-tree volume plugins, like snapshots and volume cloning.
+
+### Goals
+
+- Create a cluster operator to install the OpenStack Cinder CSI driver.
+
+- Publish the operator on OperatorHub.
+
+- Package and ship downstream images for the operator and OpenStack Cinder driver.
+
+### Non-Goals
+
+- Driver creation as it is available in [upstream](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-cinder-csi-plugin.md).
+
+- Implement CSI tooling (snapshotter, resizer, cloner). It is available in OpenShift since 4.3.
+
+## Proposal
+
+OCP ships a new cluster operator called cinder-csi-driver-operator.
+
+1. The operator installs the OpenStack Cinder CSI driver following the [recommended guidelines (currently work-in-progress)](https://github.com/openshift/enhancements/pull/139/files).
+2. The operator deploys the all the objects required by the OpenStack Cinder CSI driver:
+   2.1 A namespace called `openshift-cinder-csi-driver`.
```

Comment:
Use `openshift-cluster-csi-drivers` namespace, where all OCP CSI drivers should run (except for Manila)